### PR TITLE
FCONS-01: Added code example for each language; broke up conrefs per …

### PR DESCRIPTION
**Pull request by tonyjhillman**
…section, to ensure flexibility

### CODE REVIEW

**Comment by tonyjhillman** [comment]
I don't have an example for C yet. I notice that the C page has an additional section, plus code example, and I'm assuming that it still does need the same addition as the others.

I've added the extra example of the experimental method for Java: I assume this would be just for 5.0.

File: content/sdk/go/failure-considerations.dita [hunk 1]
```diff
@@ -1,11 +1,77 @@
...
+                replica node for a copy of the item. 
+                
+                <note>The item received from a replica node may
+                    be an older version. It is possible a newer version exists in the active node,
+                    but did not manage to get replicated before the active node went
+                    offline.
+                </note>
+            </p>
+            
+            <codeblock outputclass="language-c">var value interface
```
Comment (avsej) [inline content/sdk/go/failure-considerations.dita]: shouldn't it be `"language-go"`?

Comment (tonyjhillman) [reply content/sdk/go/failure-considerations.dita]: Yes. Though oddly, Oxygen doesn't give me that as a completion option. I can try.

**Comment by jeffrymorris** [comment]
.NET reviewed:

a) "NO NODE AVAILABLE" is specifically ResponseStatus.NodeUnavailable and an exception of type Couchbase.Core.NodeUnavailableException. I am not sure how specific you want to be though...
b) "Like timeout errors, network errors may be transient (indicative of a bad network connection). They may also be a result of a node being failed over.
Applications may retry operations which failed with network errors after waiting for a while (as in timeout errors]. “ Earlier in the doc it states that timeouts should not be retried but seem to imply that can be re-tried here? Also, ")" not "]"
c) The replica example shows only KeyNotFound as an event to trigger a replica read, however above this it states "In conditions where access to the active node is unavailable (for example, it is disconnected from the network), an application may be able to access a replica version of the item using the get-from-replica operation which queries a replica node for a copy of the item."
In that case NodeUnavailable would be returned, so there are other cases in which a replica read should be used based on the ResponseStatus that is returned by the SDK.

**Comment by tonyjhillman** [comment]
Merge targeted for DOC-S2-Nov03.

**Comment by tonyjhillman** [comment]
@jeffrymorris So, am I correct in interpreting your points (a), (b), and (c) as applicable across all languages (rather than .NET-related alone)? (Most of the text for this page is currently shared, rather than language-specific...)

**Comment by jeffrymorris** [comment]
@tonyjhillman -

a) The namespace is specific to .net, so probably the more generic description is ok.
b) Is related to the content itself + the typo.
c) should be cross sdk

**Comment by tonyjhillman** [comment]
With reference to "NodeUnavailable": I'm wondering how to handle this. Should we provide a definitive (or close to definitive) list of the responses that would potentially trigger a replica read? 

(And is it just these two, or might there be more? thanks..)

**Comment by tonyjhillman** [comment]
I'll merge and close this, and open a separate ticket on the NodeUnavailable issue.

**Comment by tonyjhillman** [comment]
DOC-2627 can now be closed.

**Comment by 1tylermitchell** [comment]
these look good but I can't review the final product very well due to the merge deleting the cb-docs-robot site - I'll see if I can get to them on the staging site.

**Comment by tonyjhillman** [comment]
Sorry..I thought you had seen previously.


Sent from my Samsung Galaxy smartphone.
-------- Original message --------From: Tyler Mitchell <notifications@github.com> Date: 15/11/2017  15:46  (GMT-08:00) To: couchbase/docs-cb4 <docs-cb4@noreply.github.com> Cc: tonyjhillman <tonyjhillman@gmail.com>, State change <state_change@noreply.github.com> Subject: Re: [couchbase/docs-cb4] FCONS-01: Added code example for each language; broke up conrefs per … (#1965) 
these look good but I can't review the final product very well due to the merge deleting the cb-docs-robot site - I'll see if I can get to them on the staging site.

—
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/couchbase/docs-cb4","title":"couchbase/docs-cb4","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/couchbase/docs-cb4"}},"updates":{"snippets":[{"icon":"PERSON","message":"@1tylermitchell in #1965: these look good but I can't review the final product very well due to the merge deleting the cb-docs-robot site - I'll see if I can get to them on the staging site."}],"action":{"name":"View Pull Request","url":"https://github.com/couchbase/docs-cb4/pull/1965#issuecomment-344767358"}}}

**Pull request merged**